### PR TITLE
[stable-2.9] VMware: Fix issue with order of changes in vmware_vcenter_statistics (61345)

### PR DESCRIPTION
**Pull request by mariolenz**
##### SUMMARY
Backporting #61345 to 2.9

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
vmware_vcenter_statistics

##### ADDITIONAL INFORMATION
Simple cherry picking.

### CODE REVIEW

**Review by goneri — APPROVED** [review]

**Comment by mariolenz** [comment]
@goneri 
> If this is the question, I absolutely trust abadger ability to find subtle Python issue :-)

I trust him, too. His objection is definitely reasonable, he's absolutely right. I was just hoping you would live with a little fragility... on the other hand: If I were in your position, I would act exactly as you did ;-)

> I understand your frustration

There's no frustration at all, I'm fine. Well, at least there's no frustration on this point. As I said: I agree with abedger, my code probably is a bit fragile.

> the VMware modules only support Py>=2.7, so it should not be to hard to implement the suggestion.

You're wrong there. Even if the VMware modules only support Py>=2.7, not being Python 2.6 compatible breaks your CI pipeline: #60288 This _is_ a bit frustrating.

@abadger Can you please have a look at PR #62288? If this is OK for you, I'd update this PR accordingly as soon as it's merged.

**Comment by abadger** [comment]
I've merged the change in devel and cherry-picked it here for you.  Waiting on shippable to pass and then I can merge this.

**Comment by abadger** [comment]
Merged for the 2.9.0rc1 release.  Thanks for putting up with my review!

**Pull request merged**
